api/children/childrenMiddleware.js

```diff
@@ -29,23 +30,44 @@ const isChildAlreadyEnrolled = async (req, res, next) => {
   }
 };
 
+const isChildParent = async (req, res, next) => {
+  const { child_id } = req.params;
+  const { profile_id } = req.profile;
+  const [parent] = await Children.findChildParent(child_id);
+  if (parent.profile_id === profile_id) {
+    next();
+  } else {
+    next({ status: 403, message: 'This is not your child please stay away!' });
```

Comment:
We _might_ want to adjust this error message to be slightly less accidentally hostile - keep in mind, we want to hand useful error messages up to the FE that don't necessarily accuse our users!